### PR TITLE
Do not error if sequence_index is numerical

### LINKED ISSUE #2079: PARSynthesizer errors during `fit` if sequence_index is numerical sdtype
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version:
* Python version:
* Operating System:

### Error Description
After #2043, we fixed an issue where `enforce_min_max_values` was by default being set to `True` for the sequence_index transformer. However, if no transformer is assigned to the sequence_index (i.e. if the sequence is already a numerical sdtype), fit now errors.

To fix, we should check that (1) a transformer has been assigned (transformer is not `None`) and (2) that the transformer has the `enforce_min_max_values` attribute (instead of adding an additional check, we could use getattr with a `False` default value in place of directly accessing the attribute)

### Steps to reproduce
```python
from sdv.datasets.demo import download_demo
from sdv.sequential import PARSynthesizer

data, metadata = download_demo('sequential', 'CMAPPSJetEngine')
s1 = PARSynthesizer(metadata)
s1.fit(data)
```
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[4], line 1
----> 1 s1.fit(data1)
      2 s1.sample(10)

File ~/Documents/SDV/sdv/single_table/base.py:471, in BaseSynthesizer.fit(self, data)
    469 self._data_processor.reset_sampling()
    470 self._random_state_set = False
--> 471 processed_data = self.preprocess(data)
    472 self.fit_processed_data(processed_data)

File ~/Documents/SDV/sdv/single_table/base.py:407, in BaseSynthesizer.preprocess(self, data)
    400     warnings.warn(
    401         'This model has already been fitted. To use the new preprocessed data, '
    402         "please refit the model using 'fit' or 'fit_processed_data'."
    403     )
    405 is_converted = self._store_and_convert_original_cols(data)
--> 407 preprocess_data = self._preprocess(data)
    409 if is_converted:
    410     data.columns = self._original_columns

File ~/Documents/SDV/sdv/sequential/par.py:286, in PARSynthesizer._preprocess(self, data)
    284 sequence_key_transformers = {sequence_key: None for sequence_key in self._sequence_key}
    285 if not self._data_processor._prepared_for_fitting:
--> 286     self.auto_assign_transformers(data)
    288 self.update_transformers(sequence_key_transformers)
    289 preprocessed = super()._preprocess(data)

File ~/Documents/SDV/sdv/sequential/par.py:267, in PARSynthesizer.auto_assign_transformers(self, data)
    265 if self._sequence_index:
    266     sequence_index_transformer = self.get_transformers()[self._sequence_index]
--> 267     if sequence_index_transformer.enforce_min_max_values:
    268         sequence_index_transformer.enforce_min_max_values = False

AttributeError: 'NoneType' object has no attribute 'enforce_min_max_values'
```

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #2079 
CU-86b0y3zdj

Ensure that if no transformer is assigned to sequence index, then do not error out.

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0y3zdj SDV - PARSynthesizer errors during `fit` if sequence_index is numerical sdtype #2079](https://app.clickup.com/t/86b0y3zdj)

**Review by frances-h — COMMENTED** [review]
Can we add an integration test as well?

**Review by frances-h — APPROVED** [review]
LGTM 👍

**Review by amontanez24 — APPROVED** [review]

File: sdv/sequential/par.py [hunk 1]
```diff
@@ -264,7 +264,9 @@ def auto_assign_transformers(self, data):
         # Ensure that sequence index does not get auto assigned with enforce_min_max_values
         if self._sequence_index:
             sequence_index_transformer = self.get_transformers()[self._sequence_index]
-            if sequence_index_transformer.enforce_min_max_values:
+            if sequence_index_transformer and getattr(
+                sequence_index_transformer, 'enforce_min_max_values', False
+            ):
```
Comment (amontanez24) [inline sdv/sequential/par.py:269]: Actually can we also add a quick unit test that has a sequence index with a transformer but not the attribute?

Comment (lajohn4747) [reply sdv/sequential/par.py:269]: Added

**Pull request merged**